### ISSUE TITLE
GDCR20: Technical Setup

### ISSUE THREAD

**Issue by caichinger** [swkWien/sessions#56]
The technical setup is still being discussed and depends on the format of the retreat.
In particular it depends on the participants, with more newcomers requiring a more prepared setup.

#55 listed the following ideas and considerations.

Current options include:
- Use zoom for everyting (can be provided by @haraldreingruber)
- Use https://gather.town/ for additional interactivity (will be tested by @dertseha and @caichinger)
- Use an already prepared GitHub repo to simplify handovers
- Use https://www.cyber-dojo.org/ to avoid handovers alltogether
- Use a collaboration plugin in an IDE to simplify sharing
- Use trello for retrospectives (and if needed, ad hoc discussions)

**Comment by gregorriegler** [comment]
I recently tested the IntelliJ CodeWithMe. The Editing and collaborating kinda works, but it lacks a lot of features such as:
- refactor move static method and other refactorings do not work on the client
- can't see code completion of peer
- client can't see commit window
- client can't commit
- connection broke accidentally
- run configuration renaming not updated on peer
- gradle test run not visible on client, junit run configuration works as a workaround

So i'd personally not use it over screensharing.

**Comment by caichinger** [comment]
> I recently tested the IntelliJ CodeWithMe. The Editing and collaborating kinda works, but it lacks a lot of features such as:
> [...]
> So i'd personally not use it over screensharing.

That is good to know, thank you for trying and sharing.

---

To add on that: When I last tried to hand screen control over using Zoom, it worked but the other person had to deal with quite annoying delays. It sounds nice in theory but in my opinion is not worth it.

**Comment by haraldreingruber** [comment]
I agree. Remote control via Zoom has a bad user experience.

VSCode LiveShare or Cyberdojo might be better.
For VSCode LiveShare, participants would need to sign up via their Microsoft or Github account ideally before the event, so that they are able to share their project.

We used Cyberdojo in a mobretreat but didn't use the prepared katas. We just chose the language and test framework as each group decided. Every test run basically creates a new version, the other participants just need to refresh with the session link to get the latest code.

**Comment by haraldreingruber** [comment]
Shall we do a short test session, with the most promising setup options?

**Comment by caichinger** [comment]
> Shall we do a short test session, with the most promising setup options?

Yes, let's do that end of next week.
Thank you for sharing your thoughts.

Cyberdojo worked nicely in Christian's dojos as well, I think this is a safe option.
Having a fixed driver and small teams (minimob?) worked well too.

For programming languages benefiting from a heavier IDE (thinking of Java), there are not so many options - it boils down to sharing the code I guess. Or am I missing something?

**Comment by haraldreingruber** [comment]
> Yes, let's do that end of next week.

Sounds good, I would be available on Thursday evening or Saturday.

**Comment by caichinger** [comment]
@dertseha and I had a short https://gather.town/ trial today and I want to give you a brief update.

TLDR; we will give it a try

The tool is very easy to use and everything worked out of the box. Having a character to navigate around a (virtual) space and interact with the environment was fun and really felt a bit like "we are in this space together".
Important aspects:
- You see each other in the environment; you get a feeling for who is here
- You communicate based on spatial proximity; however global communication is possible as well
- Specific regions in the space, so called private spaces, restrict communication to those within that space
- Interactive elements like whiteboards, posters exist and are easy to use

What we aim for:
- An environment to facilitate our retreat
- There will be private spaces for each pair/team, i.e. at the beginning of the session the pairs/teams the respective participates move into their private space
- There will be an area for retrospectives, i.e. at the end of each session the pairs/teams move to that area
- There will be a podium for general announcements
- Maybe there will be games for the lunch break

Of course, some details still need to be sorted out and final decisions made in the end but at the moment it looks like a really nice interactive way to facilitate our event.

@dertseha will try to prepare/build this space. @dertseha please comment as you see fit. 

---

*We still keep (and need) Zoom as backup option. There is Zoom call integration as well, maybe that's useful; we do not know yet.*

**Comment by haraldreingruber** [comment]
Sounds great, looking forward to trying out gather.town!

**Comment by codecop** [comment]
I have forked https://github.com/swkWien/kata-bootstraps-gdcr20 and add you (group) as maintainer and invited all ppl currently registered. This can be used to share code for pairs.

**Comment by haraldreingruber** [comment]
That's awesome @codecop. I think our current plan is to basically let the pairs/teams decide on their own. But we will suggest to use cyber-dojo.org, at least it will be the only one where we will offer technical support...

That idea is probably not yet set in stone, and I guess we will talk about it during the next meeting.

**Comment by caichinger** [comment]
Thank you for setting up a repo and inviting the participants.

Yes, cyber-dojo.org is what we suggest as ready-made-solution but having the repo with members already added will speed things up for those who want to run their local setup.

[WIP: Intro slides with tech setup](https://docs.google.com/presentation/d/1a5g_I00qTFCQvblCVSl1-EdFVB1SKoRX4eKWuB9Go0I/edit#slide=id.ga6eb3a7f59_0_29)

[WIP: Working document with all links](https://docs.google.com/document/d/1Vy79tH-ZhY4EZSaiOoAScReCWtLBJX2fnywf6Odhvc4/edit?usp=sharing)

Final discussion on Thursday. Closing here.

Thank you all!
